### ISSUE TITLE
Properly support horizontal scrolling in commits graph

### ISSUE THREAD

**Issue by esotericist** [JetpackDuba/Gitnuro#177]
**Is your feature request related to a problem? Please describe.**
for a user with impaired visibility, it's sometimes necessary to set scaling high enough that not everything fits cleanly no matter how you adjust things. in a situation like that, horizontal scrolling becomes necessary.

**Describe the solution you'd like**
i'd like if more elements of the ui would manifest a scrollbar when appropriate, and any ui element with a scrollbar should properly respond to horizontal scroll events.

**Describe alternatives you've considered**
this is pretty basic ux stuff, enough that i verged on filing it as a bug. i don't think a reasonable alternative exists.

**Additional context**

panes narrowed excessively for demonstrative purposes:
![image](https://github.com/JetpackDuba/Gitnuro/assets/1569754/da80e803-f0f4-4cdd-8d99-563f89e44fb7)

the graph shows a horizontal scrollbar, but does not respond to horizontal scroll events. 
the commit list has no scrollbar, and obviously also does not respond to horizontal scroll events (since it cannot scroll horizontally).

also as a note, the tooltip for the graph likes to appear over the scrollbar, consequently pulling focus off of the scrollbar, inhibiting clicking. this definitely is a bug, and could be filed separately, but i suspected it would be more likely to be addressed as part of deeper changes to the panel behavior. if you want it separately filed i can do that.

![image](https://github.com/JetpackDuba/Gitnuro/assets/1569754/969606be-6ad2-41b7-bead-94cc8621b803)
(username and 'email' address marked out for propriety, even though it's a github placeholder on a public repo. idk, i felt like a jerk not doing that)

**Comment by JetpackDuba** [comment]
For some reason the even is not registered (it should be done by default). I'll do some research about it.

**Comment by JetpackDuba** [comment]
I found that due to how views are stacked, the scroll event was consumed by the view at the top instead of the one at the bottom (which has the scroll information). I've applied a fix which passes properly the information to the scroll handler.

I've also included a fix to the tooltip you reported, it will be shown on the side instead.
![hover](https://github.com/JetpackDuba/Gitnuro/assets/16917468/5b7f78b0-6eb5-4300-be69-9e45f97ff923)


Both fixes will be available in the next release :smile:
